rdsbroker/parameter_groups_source_test.go

```diff
@@ -0,0 +1,294 @@
+				DBName:                      "",
+				PreferredBackupWindow:       "",
+				PreferredMaintenanceWindow:  "",
+				SkipFinalSnapshot:           nil,
+				RestoreFromLatestSnapshotOf: nil,
+				Extensions:                  nil,
+			}
+
+			logger := lager.NewLogger("rdsbroker_test")
+			gingkoSink := lager.NewWriterSink(GinkgoWriter, lager.INFO)
+			logger.RegisterSink(gingkoSink)
+			testSink := lagertest.NewTestSink()
+			logger.RegisterSink(testSink)
+
+			rdsFake = &fakes.FakeRDSInstance{}
+			parameterGroupSource = NewParameterGroupSource(config, rdsFake, logger)
+		})
+
+		It("returns an error when the RDS api returns an error other than not found", func() {
+			rdsError := awserr.New(rds.ErrCodeDBClusterAlreadyExistsFault, "not found", nil)
+			rdsFake.GetParameterGroupReturns(nil, rdsError)
+
+			_, err := parameterGroupSource.SelectParameterGroup(servicePlan, provisionDetails)
+			Expect(err).To(HaveOccurred())
+		})
+
+		Describe("when the parameter group exists", func() {
+			BeforeEach(func() {
+				rdsFake.GetParameterGroupReturns(&rds.DBParameterGroup{
+					DBParameterGroupArn:    aws.String("aws:arn:::db-parameter-group"),
+					DBParameterGroupFamily: aws.String("postgres"),
+					DBParameterGroupName:   aws.String("rdsbroker-postgres10-envname"),
+					Description:            aws.String("rdsbroker-postgres10-envname"),
+				}, nil)
+			})
+
+			It("does not attempt to create the group", func() {
+				parameterGroupSource.SelectParameterGroup(servicePlan, provisionDetails)
+				createGroupCalls := rdsFake.CreateParameterGroupCallCount()
+
+				Expect(createGroupCalls).To(Equal(0))
+			})
+
+			It("returns the group name", func() {
+				name, _ := parameterGroupSource.SelectParameterGroup(servicePlan, provisionDetails)
+				Expect(name).To(Equal("rdsbroker-postgres10-envname"))
+			})
+		})
+
+		Describe("when the parameter group does not exist", func() {
+			BeforeEach(func() {
+				rdsFake.GetParameterGroupReturns(nil, errors.New(rds.ErrCodeDBParameterGroupNotFoundFault+": errMsg"))
+			})
+
+			It("attempts to create the group", func() {
+				rdsFake.CreateParameterGroupReturns(nil)
+
+				parameterGroupSource.SelectParameterGroup(servicePlan, provisionDetails)
+
+				Expect(rdsFake.CreateParameterGroupCallCount()).To(Equal(1))
+				createGroupCall := rdsFake.CreateParameterGroupArgsForCall(0)
+				Expect(aws.StringValue(createGroupCall.DBParameterGroupName)).To(Equal("rdsbroker-postgres10-envname"))
+			})
+
+			It("returns an error if creating the parameter group fails", func() {
+				createError := awserr.New(rds.ErrCodeDBParameterGroupAlreadyExistsFault, "exists", nil)
+				rdsFake.CreateParameterGroupReturns(createError)
+
+				_, err := parameterGroupSource.SelectParameterGroup(servicePlan, provisionDetails)
+
+				Expect(err).To(HaveOccurred())
+			})
+
+			DescribeTable(
+				"it modifies the created parameter group",
+				func(paramName *string, paramValue *string, applyMethod *string) {
```

Comment:
This should not be a table-type test as that's used for different input + expectation sets. In this case you would just want to list all Expect() calls for all properties.